test/server/cards/03_TWI/events/InDefenseOfKamino.spec.ts

```diff
@@ -0,0 +1,57 @@
+describe('In Defense of Kamino', function () {
+    integration(function (contextRef) {
+        describe('In Defense of Kamino\'s ability', function () {
+            it('should give each friendly Republic unit Restore 2 and: "When Defeated: Create a Clone Trooper token."', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['in-defense-of-kamino'],
+                        groundArena: ['advanced-recon-commando', 'ryloth-militia'],
+                        spaceArena: ['republic-arc170']
+                    },
+                    player2: {
+                        groundArena: ['b2-legionnaires'],
+                        spaceArena: ['gladiator-star-destroyer']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.inDefenseOfKamino);
+                expect(context.player2).toBeActivePlayer();
+
+                context.player2.clickCard(context.gladiatorStarDestroyer);
```

Comment:
Can start the base with damage on it like this:

```ts
base: { card: 'echo-base', damage: 5 }
```

Reply:
gotcha, done

Reply:
Is it better to do something like that and use passAction() or simulate actual gameplay?